rcl/src/rcl/time.c

```diff
@@ -77,257 +72,184 @@ rcl_get_ros_time(void * data, rcl_time_point_value_t * current_time)
-    rcl_default_system_time_source = (rcl_time_source_t *)calloc(1, sizeof(rcl_time_source_t));
-    rcl_ret_t retval = rcl_system_time_source_init(rcl_default_system_time_source);
-    if (retval != RCL_RET_OK) {
-      return NULL;
-    }
-  }
-  return rcl_default_system_time_source;
-}
-
-rcl_ret_t
-rcl_set_default_ros_time_source(rcl_time_source_t * process_time_source)
-{
-  RCL_CHECK_ARGUMENT_FOR_NULL(
-    process_time_source, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
-  if (rcl_default_ros_time_source) {
-    free(rcl_default_ros_time_source);
-  }
-  rcl_default_ros_time_source = process_time_source;
-  return RCL_RET_OK;
-}
-
-rcl_ret_t
 rcl_difference_times(
   rcl_time_point_t * start, rcl_time_point_t * finish, rcl_duration_t * delta)
 {
-  if (start->time_source->type != finish->time_source->type) {
+  if (start->clock_type != finish->clock_type) {
     RCL_SET_ERROR_MSG(
-      "Cannot difference between time points with time_sources types.",
+      "Cannot difference between time points with clocks types.",
       rcl_get_default_allocator());
     return RCL_RET_ERROR;
   }
   if (finish->nanoseconds < start->nanoseconds) {
     rcl_time_point_value_t intermediate = start->nanoseconds - finish->nanoseconds;
     delta->nanoseconds = -1 * (int) intermediate;
+  } else {
+    delta->nanoseconds = (int)(finish->nanoseconds - start->nanoseconds);
   }
-  delta->nanoseconds = (int)(finish->nanoseconds - start->nanoseconds);
   return RCL_RET_OK;
 }
 
 rcl_ret_t
-rcl_time_point_get_now(rcl_time_point_t * time_point)
+rcl_clock_get_now(rcl_clock_t * clock, rcl_time_point_t * time_point)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(time_point, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
-  if (time_point->time_source && time_point->time_source->get_now) {
-    return time_point->time_source->get_now(time_point->time_source->data,
+  if (clock->type && clock->get_now) {
+    return clock->get_now(clock->data,
              &(time_point->nanoseconds));
   }
   RCL_SET_ERROR_MSG(
-    "time_source is not initialized or does not have get_now registered.",
+    "clock is not initialized or does not have get_now registered.",
     rcl_get_default_allocator());
   return RCL_RET_ERROR;
 }
 
 rcl_ret_t
-rcl_enable_ros_time_override(rcl_time_source_t * time_source)
+rcl_enable_ros_time_override(rcl_clock_t * clock)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(time_source, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
-  if (time_source->type != RCL_ROS_TIME) {
+  RCL_CHECK_ARGUMENT_FOR_NULL(clock, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
+  if (clock->type != RCL_ROS_TIME) {
     RCL_SET_ERROR_MSG(
       "Time source is not RCL_ROS_TIME cannot enable override.", rcl_get_default_allocator())
     return RCL_RET_ERROR;
   }
-  rcl_ros_time_source_storage_t * storage = \
-    (rcl_ros_time_source_storage_t *)time_source->data;
+  rcl_ros_clock_storage_t * storage = \
+    (rcl_ros_clock_storage_t *)clock->data;
```

Comment:
This line looks short enough to not need to be wrapped. Same thing on a lot of similar lines below